linkerd/app/gateway/src/gateway.rs

```diff
@@ -63,6 +78,23 @@ where
             None => return Gateway::BadDomain(http.target.name().clone()),
         };
 
+        // Create an outbound target using the endpoint from the profile.
+        if let Some((addr, metadata)) = profile.endpoint() {
+            debug!("Creating outbound endpoint");
+            let svc = self
+                .outbound
+                .new_service(svc::Either::B(outbound::http::Endpoint::from((
+                    http.version,
+                    outbound::tcp::Endpoint::from_metadata(
+                        addr,
+                        metadata,
+                        self.no_tls_reason,
```

Comment:
I don't think we actually need to wire `no_tls_reason` through the constructor. `no_tls_reason` is really to handle the case when TLS is disabled on a workload so we don't report "not provided by service discovery" when we should report "identity disabled".

But identity **can't** be disabled on the gateway. We require ALPN for protocol negotiation, etc. Just above this we enforce this:
https://github.com/linkerd/linkerd2-proxy/blob/5ef0c35ccea0719b322bffe9fb668dc44b049073/linkerd/app/gateway/src/gateway.rs#L72-L75

So, we should probably omit `no_tls_reason` on the gateway and just hardcode the NotProvidedByServiceDiscovery reason here.

Reply:
This makes sense. I was wondering whether we'd have a `None` value at any point or if we have a default value for this (so we can avoid making `no_tls_reason` public and pass it through). It's demystified now!